src/test/java/com/spotify/futures/CompletableFuturesTest.java

```diff
@@ -467,6 +490,78 @@ public void ctor_preventInstantiation() throws Exception {
     ctor.newInstance();
   }
 
+  @Test
+  public void poll_done() throws Exception {
+    final Supplier<Optional<String>> supplier = () -> Optional.of("done");
+    final CompletableFuture<String> future = poll(supplier, Duration.ofMillis(2), executor);
+
+    executor.runNextPendingCommand();
+    assertThat(future, completesTo("done"));
+  }
+
+  private static class ExternalTask {
```

Comment:
I wonder if this could be replaced with something like:

```java
Supplier<Optional<String>> task = Mockito.mock(Supplier.class);
Mockito.when(task.get()).thenReturn(Optional.empty(), Optional.of("done");
```

(Not important though, I still approve of this)